src/libraries/System.Runtime.Caching/src/System/Runtime/Caching/HostFileChangeMonitor.cs

```diff
@@ -95,6 +95,7 @@ private static void InitFCN()
                 {
                     fcn = host.GetService(typeof(IFileChangeNotificationSystem)) as IFileChangeNotificationSystem;
                 }
+#pragma warning disable IDE0074 // Use compound assignment
```

Comment:
I think there should also be a dotnet/roslyn issue for this. It shouldn't be recommending compound assignment if there are ifdefs in the way.